### ISSUE TITLE
Correct and wrong RRSIG in one answer - Number of labels RRset vs RRSIG label field

### ISSUE THREAD

**Issue by JuergenAuer** [dnsviz/dnsviz#86]
Asking A + DNSSEC a name server may send more then one RRSIG.

Before validating a signature, a name server must check some conditions, see https://datatracker.ietf.org/doc/html/rfc4035#section-5.3.1

The RRSIG and the RRset must have the same owner, class, the types must be the same, the time must be correct. And:

> The number of labels in the RRset owner name MUST be greater than or equal to the value in the RRSIG RR's Labels field.

If an A-RR of `www.example.com` is validated, the RRSIG label field may be 3 or 2. If 2, there isn't an own A-RR with `www.example.com`. Instead, there is an A-RR with `*.example.com`.

There is a domain (found in my own tool), that sends three RRSIG. Two with `www.example.com` as owner, one with `example.com`.

One of the www-RRSIG has label = 3, the other has label = 5. The signature of the first is valid, the signature of the second is invalid.

DNSVIZ lists that second result and says, the A-RR isn't validated.

But it should list the correct RRSIG and should ignore / not check the signature of a RRSIG with label = 5.

**Comment by cdeccio** [comment]
I've pushed some code to address this.  Please note that as an analytical tool, DNSViz displays errors or warnings for things that aren't "right" - even if those things will not ultimately result in a validation error by a validating resolver.  For example, DNSViz [checks whether the TTL of the RRSIG or RRset exceeds the value of the original TTL field](https://github.com/dnsviz/dnsviz/blob/master/dnsviz/analysis/status.py#L246).  A validator will handle this behavior just fine - but it is still wrong.  And more importantly it is a sign that something else might be wrong.  So showing the error helps the developer/engineer/user track down an issue potentially *before* it becomes an issue.

Same here.  There is no good reason why the value of the labels field of the RRSIG should exceed the label count of the RRset owner name.  As long there is another RRSIG that doesn't have the same issues, then a validator *should* handle this just fine.  But the behavior is wrong, so DNSViz will flag it.

In summary, I've added a check to compare the value of the labels field of the RRSIG to the label count of the RRset owner name, and throw an error if it is greater.  It is currently sitting in the `check_label_count` branch, and I would appreciate a test 😄
